src/main/java/org/wecancodeit/pantryplus/Cart.java

```diff
@@ -40,17 +40,23 @@ public LineItem getLineItemByProductId(long productId) {
 	public int getCartQuantity() {
 		int totalQuantity = 0;
 		for (LineItem item : lineItems) {
-			totalQuantity += item.getQuantity();
+			if (item instanceof CountedLineItem) {
+				totalQuantity += ((CountedLineItem) item).getQuantity();
+			}
 		}
 		return totalQuantity;
 	}
 
+	@SuppressWarnings("unused")
 	public int getLineItemQuantityByProductId(long productId) {
 		LineItem lineItem = getLineItemByProductId(productId);
-		if (lineItem == null) {
-			return 0;
+		if (lineItem instanceof CountedLineItem) {
+			if (lineItem == null) {
+				return 0;
+			}
```

Comment:
This if statement: (lineItem == null) should be outside of (lineItem instanceof CountedLineItem) and should come before it.